Compiling with C99 not taking effect when building using Gradle [5669]

While building on RedHat/CentOS, where the default compiler is `gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-36)` the default C standard is c90 and gives building the generator issues, even with version 1.1.1 of the client. However, I was able to get the generator to build by doing one of the following things:

- Setting an environment variable to change the default c flags, which filters into cmake.
  - `export CFLAGS='-std=c99'`
- Modifying [MicroXRCEDDSGenIntegrationTest.java](https://github.com/eProsima/Micro-XRCE-DDS-Gen/blob/master/src/test/java/com/eprosima/uxr/integration/MicroXRCEDDSGenIntegrationTest.java) to include `-DCMAKE_C_FLAGS='-std=c99'` when configuring cmake.

Gradle build error:
```
[...]
com.eprosima.uxr.integration.MicroXRCEDDSGenIntegrationTest > runTests STANDARD_OUT
    Executing command: mkdir -p build/test/integration
    Executing command: rm -rf Micro-XRCE-DDS-Client
    Executing command: git clone -b master https://github.com/eProsima/Micro-XRCE-DDS-Client.git
    Executing command: mkdir build
    Executing command: cmake .. -DCMAKE_INSTALL_PREFIX=install
    Executing command: make
    Executing command: make install


    >>> BASIC TEST: PREPARE...
    Done!
        RESULT: OK!


    >>> BASIC TEST: GENERATE...
    Executing command: java -jar share/microxrcedds/microxrceddsgen.jar -replace -example -test -d build/test/integration/idls/basic thirdparty/IDL-Parser/test/idls/basic.idl
        RESULT: OK!


    >>> BASIC TEST: CONFIGURE...
    Executing command: cmake .. -DCMAKE_PREFIX_PATH=/Micro-XRCE-DDS-Gen/build/test/integration/Micro-XRCE-DDS-Client/build/install
        RESULT: OK!


    >>> BASIC TEST: COMPILE...
    Executing command: make
    Scanning dependencies of target basicSerializationTest
    [ 11%] Building C object CMakeFiles/basicSerializationTest.dir/basicSerializationTest.c.o
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c:1:1: warning: C++ style comments are not allowed in ISO C90 [enabled by default]
     // Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
     ^
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c:1:1: warning: (this will be reported only once per input file) [enabled by default]
    In file included from /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c:22:0:
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basic.h:1:1: warning: C++ style comments are not allowed in ISO C90 [enabled by default]
     // Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
     ^
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basic.h:1:1: warning: (this will be reported only once per input file) [enabled by default]
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c: In function 'testBasic':
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c:95:5: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
         uint8_t buffer[BUFFER_SIZE];
         ^
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c:104:5: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
         uint32_t Basic_serialization_topic_size = Basic_size_of_topic(&Basic_serialization_topic, 0);
         ^
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c:122:5: error: 'for' loop initial declarations are only allowed in C99 mode
         for(int i = 0; i < buffer_length; ++i)
         ^
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c:122:5: note: use option -std=c99 or -std=gnu99 to compile your code
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c: In function 'main':
    /Micro-XRCE-DDS-Gen/build/test/integration/idls/basic/basicSerializationTest.c:136:5: warning: implicit declaration of function 'time' [-Wimplicit-function-declaration]
         srand((unsigned) time(NULL));
         ^
    make[2]: *** [CMakeFiles/basicSerializationTest.dir/basicSerializationTest.c.o] Error 1
    make[1]: *** [CMakeFiles/basicSerializationTest.dir/all] Error 2
    make: *** [all] Error 2
        RESULT: ERROR

com.eprosima.uxr.integration.MicroXRCEDDSGenIntegrationTest > runTests SKIPPED

> Task :test FAILED
:test (Thread[Task worker for ':' Thread 8,5,main]) completed. Took 13.062 secs.
[...]
```

--- a/src/main/resources/com/eprosima/uxr/idl/templates/CMakeLists.stg
+++ b/src/main/resources/com/eprosima/uxr/idl/templates/CMakeLists.stg
@@ -26,13 +26,6 @@ project(client_examples)
 find_package(microcdr REQUIRED)
 find_package(microxrcedds_client REQUIRED)
 
-# Warnings
-if(MSVC OR MSVC_IDE)
-    set(CMAKE_C_FLAGS "\${CMAKE_C_FLAGS} /W4 /wd4700 /wd4996 /wd4820 /wd4255 /wd4668")
-else()
-    set(CMAKE_C_FLAGS "\${CMAKE_C_FLAGS} -pedantic -fstrict-aliasing -Wall -Wcast-align -Wshadow")
-endif()
-
 set(SRCS
     $solution.projects : {$it.name$.c}; separator="\n"$
     )
@@ -44,16 +37,48 @@ execs(project, libraries, examples, test) ::= <<
 # $project.name$ Subscriber
 add_executable($project.name$Subscriber $project.name$Subscriber.c \${SRCS})
 target_link_libraries($project.name$Subscriber microxrcedds_client)
+set_target_properties($project.name$Subscriber PROPERTIES
+    C_STANDARD
+        99
+    C_STANDARD_REQUIRED
+        YES
+    )
+target_compile_options($project.name$Subscriber
+    PRIVATE
+        \$<\$<C_COMPILER_ID:MSVC>:/W4 /wd4700 /wd4996 /wd4820 /wd4255 /wd4668>
+        \$<\$<C_COMPILER_ID:GNU>:-pedantic -fstrict-aliasing -Wall -Wcast-align -Wshadow>
+    )
 
 # $project.name$ Publisher
 add_executable($project.name$Publisher $project.name$Publisher.c \${SRCS})
 target_link_libraries($project.name$Publisher microxrcedds_client)
-
+set_target_properties($project.name$Publisher PROPERTIES
+    C_STANDARD
+        99
+    C_STANDARD_REQUIRED
+        YES
+    )
+target_compile_options($project.name$Publisher
+    PRIVATE
+        \$<\$<C_COMPILER_ID:MSVC>:/W4 /wd4700 /wd4996 /wd4820 /wd4255 /wd4668>
+        \$<\$<C_COMPILER_ID:GNU>:-pedantic -fstrict-aliasing -Wall -Wcast-align -Wshadow>
+    )
 
 $if(test)$
 # $project.name$ Serialization Test
 add_executable($project.name$SerializationTest $project.name$SerializationTest.c \${SRCS})
 target_link_libraries($project.name$SerializationTest microxrcedds_client)
+set_target_properties($project.name$SerializationTest PROPERTIES
+    C_STANDARD
+        99
+    C_STANDARD_REQUIRED
+        YES
+    )
+target_compile_options($project.name$SerializationTest
+    PRIVATE
+        \$<\$<C_COMPILER_ID:MSVC>:/W4 /wd4700 /wd4996 /wd4820 /wd4255 /wd4668>
+        \$<\$<C_COMPILER_ID:GNU>:-pedantic -fstrict-aliasing -Wall -Wcast-align -Wshadow>
+    )
 
 $endif$
 >>

